clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1139,6 +1295,31 @@ static void populateIntHeader(SYCLIntegrationHeader &H, const StringRef Name,
       if (ArgTy->isStructureOrClassType()) {
         populateHeaderForWrappedAccessors(ArgTy, Offset);
       }
+    } else if (ArgTy->isConstantArrayType()) {
+      auto CAT = cast<ConstantArrayType>(ArgTy);
+      auto ET = CAT->getElementType();
+      if (Util::isSyclAccessorType(ET)) {
+        size_t NumElements = static_cast<size_t>(CAT->getSize().getZExtValue());
+        const auto *Accessor = ET->getAsCXXRecordDecl();
+        ASTContext &AccessorCtx = Accessor->getASTContext();
+        const ASTRecordLayout &AccessorLayout =
+            AccessorCtx.getASTRecordLayout(Accessor);
+        auto AccessorSize = AccessorCtx.toBits(AccessorLayout.getSize()) / 8;
+        // Start with offset of array, i.e., offset of first element of array
+        uint64_t OffsetInArray = Offset;
+        // Treat each element of accessor array as a separate item
+        // Create a parameter descriptor for it
+        for (size_t I = 0; I < NumElements; ++I) {
+          populateHeaderForAccessor(ET, OffsetInArray);
+          OffsetInArray += AccessorSize;
```

Comment:
After lines 1231-1253 it's déjà vu again

Reply:
In this case, there was somewhat more similarity in the actions done in the two cases, and this code has been factored out into a separate function, called twice.